src/commands/report.rs

```diff
@@ -0,0 +1,56 @@
+use crate::{
+    http,
+    reporter::{self, Report},
+    settings,
+};
+
+use anyhow::Result;
+use cloudflare::framework::{
+    apiclient::ApiClient,
+    endpoint::{Endpoint, Method},
+    response::ApiFailure,
+};
+
+struct ErrorReport(Report);
+
+impl Endpoint<(), (), Report> for ErrorReport {
+    fn method(&self) -> Method {
+        Method::Post
+    }
+
+    fn path(&self) -> String {
+        "wrangler/errors".into()
+    }
+
+    fn body(&self) -> Option<Report> {
+        Some(self.0.clone())
+    }
+}
+
+pub fn run(log: Option<&str>) -> Result<()> {
+    let user = settings::global_user::GlobalUser::new()?;
+    let report = reporter::read_log(log)?;
+    let client = http::cf_v4_client(&user)?;
+    client
+        .request(&ErrorReport(report))
+        .map(|_| ())
+        .map_err(|e| match e {
+            ApiFailure::Error(code, _) => {
+                // a 409 Conflict will be returned if we have detected that a report has already
+                // been submitted.
+                if code == 409 {
+                    return anyhow::anyhow!(
+                        "we have already received this report, thank you for submitting!"
+                    );
+                }
+
+                anyhow::format_err!(
+                    "submission failed! please verify your credentials and try again. (status: {})",
+                    code
+                )
```

Comment:
To capture this error better, you can use [`Context`](https://docs.rs/anyhow/1.0.40/anyhow/trait.Context.html) to provide a custom error message and then just use `anyhow!(e)` to have anyhow keep track of the original error's source, stack trace, etc.

Reply:
You should look at the anyhow docs for the difference between passing a `std::error::Error` implementor vs a `std::fmt::Display` implementor to `anyhow!`, passing an `Error` preserves its original stack trace, etc. I know we're not capturing this right now, but eventually we might.

Reply:
good call. going to leave this as-is for now, since the errors it can use in the context _might_ only come from the `ApiErrors` struct, 
```rust
#[derive(Deserialize, Debug, Default)]
pub struct ApiErrors {
    #[serde(flatten)]
    pub other: HashMap<String, JValue>,
    pub errors: Vec<ApiError>,
}
```

and there's just a whole lot there that i'm not trying to jump into right now 🤣 .. there's an `Error` impl on it, but idk

I'll leave a TODO with some detail to look into it later.